rapid7/metasploit-framework
# rConfig <= 3.9.6  - Authenticated RCE

## Vulnerable Application

### Description

This module allows an attacker with a privileged rConfig account to start a reverse shell due to an
arbitrary file upload vulnerability in `/lib/crud/vendors.crud.php`. Then, the uploaded payload can be
triggered by a call to `images/vendor/<payload_file>.php`

### Installation

Vulnerable versions of rConfig can be downloaded from [here](https://www.cacti.net/downloads/). Then, help yourself with [this](https://help.rconfig.com/gettingstarted/installation) installation guide. 
You can also use this [docker file](https://hub.docker.com/r/libyerman/rconfig) (as long as it is not updated and remains a 3.9.6 version of rConfig)

## Verification Steps

List the steps needed to make sure this thing works

1. Start `msfconsole`
2. `use exploit/linux/http/rconfig_authenticated_rce`
3. `set USERNAME <admin_username>`
4. `set PASSWORD <admin_password>`
5. `set TARGETURI <base_path_rconfig>` if the base path of rConfig web server is different from `/`
6. `check` to check if the targeted rConfig server is vulnerable
7. `run` the module to exploit the vulnerability and start a reverse shell

## Options

### USERNAME

Set the USERNAME of your admin account.

### PASSWORD

Set the PASSWORD of your admin account.

## Scenarios

This module was successfully tested on CentOS 7 with rConfig 3.9.6. See the following output :

![image](https://user-images.githubusercontent.com/84333864/122404287-5e9bd800-cf7f-11eb-9de6-bf220792fe13.png)
